carla/models/catalog/catalog.py

```diff
@@ -219,16 +242,13 @@ def predict_proba(self, x):
         input = self.perform_pipeline(x) if self._use_pipeline else x
 
         if self._backend == "pytorch":
-            class_1 = 1 - self._model.forward(input).detach().numpy().squeeze()
-            class_2 = self._model.forward(input).detach().numpy().squeeze()
-
-            # For single prob prediction it happens, that class_1 is casted into float after 1 - prediction
-            # Additionally class_1 and class_2 have to be at least shape 1
-            if not isinstance(class_1, np.ndarray):
-                class_1 = np.array(class_1).reshape(1)
-                class_2 = class_2.reshape(1)
+            class_1 = 1 - self.predict(input)
+            class_2 = self.predict(input)
 
-            return np.array(list(zip(class_1, class_2)))
+            if torch.is_tensor(class_1):
+                return torch.cat((class_1, class_2), dim=1)
+            else:
+                return np.array(list(zip(class_1, class_2))).reshape((-1, 2))
```

Comment:
Looks very complicated haha. Are these operations generic for any pytorch model that doesn't handle with tensors?

Reply:
Keeping the output in the correct shape is really a pain, and I haven't found a better method at the moment. 
Maybe there is a better way if we use a different pytorch model. I thought we spoke some time ago about retraining at a later point.